src/InteractiveWindow/Editor/InteractiveWindow.UIThreadOnly.cs

```diff
@@ -631,7 +632,77 @@ public void AddInput(string command)
                 FinishCurrentSubmissionInput();
             }
 
-            private void AppendUncommittedInput(string text)
+            /// <summary>Implements <see cref="IInteractiveWindow2.AddToHistory"/>.</summary>
+            public void AddToHistory(string command) 
+            {
+                if (string.IsNullOrEmpty(command)) 
+                {
+                    return;
+                }
+
+
+                var historyBuffer = _factory.CreateAndActivateBuffer(_window);
```

Comment:
Looking at the implementation of the VS implementation of CreateAndActivateBuffer it seems that this is unnecessary for history buffer - it's wiring up undo manager and command filters. 

Doing this should be sufficient:

            IContentType contentType;
            if (!window.Properties.TryGetProperty(typeof(IContentType), out contentType))
            {
                contentType = _contentTypeRegistry.GetContentType("text");
            }

            var historyBuffer = _textBufferFactoryService.CreateTextBuffer(contentType);

Also, I think we should set properties:

buffer.Properties.AddProperty(typeof(IInteractiveEvaluator), Evaluator);
buffer.Properties.AddProperty(typeof(InteractiveWindow), _window);

they are set on all language buffers.